librz/core/golang.c

```diff
@@ -1605,14 +1612,81 @@ static ut32 golang_recover_string_riscv64(GoStrRecover *ctx) {
 	return 4;
 }
 
+// Sometimes the data-structures has strings, but these are stored in tables where
+// the first offset is always the pointer to the squashed strings and the next word
+// is the size of the string
+static void core_recover_golang_strings_from_data_pointers(RzCore *core, GoStrRecover *ctx) {
+	rz_core_notify_begin(core, "Recovering go strings from bin maps");
+
+	RzAnalysis *analysis = core->analysis;
+	const ut32 word_size = analysis->bits / 8;
+	RzListIter *iter;
+	RzBinMap *map;
+	ut8 *buffer = NULL;
+	ut64 string_addr, string_size;
+	RzBinObject *object = rz_bin_cur_object(core->bin);
+	RzList *map_list = object ? rz_bin_object_get_maps(object) : NULL;
+	if (!map_list) {
+		RZ_LOG_ERROR("Failed to get the RzBinMap list\n");
+		goto end;
+	}
+
+	buffer = malloc(GO_MAX_TABLE_SIZE);
+	if (!buffer) {
+		RZ_LOG_ERROR("Failed to allocate table buffer\n");
+		goto end;
+	}
+
+	rz_list_foreach (map_list, iter, map) {
+		if (!rz_bin_map_is_data(map) || map->psize < (word_size * 2)) {
+			continue;
+		}
+
+		ut64 current = map->vaddr;
+		ut64 end = map->vaddr + map->psize;
+
+		do {
+			size_t length = RZ_MIN((end - current), GO_MAX_TABLE_SIZE);
+			if (length < (word_size * 2)) {
+				break;
+			}
+
+			if (0 > rz_io_nread_at(core->io, current, buffer, length)) {
```

Comment:
this looks a bit strange TBH. We usually write `if (xxxxx(....) < 0)` rather than `0 > xxxx`.

Reply:
sure, let me patch that.